Source/Scene/ArcGisMapServerImageryProvider.js

```diff
@@ -311,14 +400,20 @@ define([
         },
 
         /**
-         * Gets the ArcGIS token used to authenticate with the ArcGis MapServer service.
+         * The ArcGIS token used to authenticate with the ArcGis MapServer service.
          * @memberof ArcGisMapServerImageryProvider.prototype
          * @type {String}
-         * @readonly
          */
         token : {
             get : function() {
                 return this._token;
+            },
+            set : function(token) {
+                delete this._token;
+
+                if (defined(token)) {
+                    this._token = token;
+                }
```

Comment:
I can't think of any reason to delete the token and then only add it if it's defined.  A simple `this._token = token` will do, and will make `_token` undefined if the new token is undefined.  There are very few places where it's a good idea to use delete.

Reply:
>  There are very few places where it's a good idea to use delete.
Is there any reason behind this.

Have changed this.

This design was put in place so as not to pollute this._token (i.e. so that the class state would not have ._token defined if it had been defined and then undefined exactly the same as when it was never defined). Can't imagine any practical differences between the two implementation, but would expect `for (var key in obj) { console.log(key); }` to differ for the two cases.

Reply:
Why do you want `_token` to not exist when it's not used?  In a statically-typed language (e.g. Java), you'd declare a field for the token and you wouldn't try to change the type depending on whether the field was used or not, right?  But that's essentially what you're doing by deleting the field.  You're forcing the browser to re-allocate the object without the field, or perhaps drop the whole object into (slower) hash mode.  Admittedly performance is not much of a concern here, but there's no reason to make the browser do extra work when `_token` should just be a declared field on the object like all the other fields.

Reply:
Matt and I described this in more detail in our chapter in WebGL Insights.  Check out section 4.3.1 here: http://webglinsights.github.io/downloads/WebGL-Insights-Chapter-4.pdf